abra_core/src/vm/vm_test.rs

```diff
@@ -578,4 +578,36 @@ mod tests {
         let expected = Value::Obj(Obj::StringObj { value: Box::new("Save the Cheerleader, Save the World!".to_string()) });
         assert_eq!(expected, result);
     }
+
+    #[test]
+    fn interpret_func_invocation_recursion() {
+        let input = "\
+          func fib(n: Int): Int {\n\
```

Comment:
Fibonacci AKA how you know you've created a real programming language implementation